client/components/create-profile-form.jsx

```diff
@@ -23,10 +24,22 @@ export default class CreateProfileForm extends React.Component {
   }
 
   componentDidMount() {
-    const { userId } = this.context.user;
-    this.setState({
-      userId
-    });
+    const { token } = this.context;
+    fetch('/api/users', {
+      headers: {
+        'X-Access-Token': token
+      }
+    })
+      .then(res => res.json())
+      .then(result => {
+        const { userName, currentWeight, profilePictureUrl } = result[0];
```

Comment:
The server shouldn't force the client to "unpack" what it wants. Have your server send an object instead of an array.